plugins/by-name/fidget/default.nix

```diff
@@ -34,650 +28,424 @@ let
       '';
 
-      warnStyle = helpers.mkNullOrOption types.str ''
+      warn_style = lib.nixvim.mkNullOrOption types.str ''
         Style used to highlight warn item annotes.
       '';
 
-      errorStyle = helpers.mkNullOrOption types.str ''
+      error_style = lib.nixvim.mkNullOrOption types.str ''
         Style used to highlight error item annotes.
       '';
 
-      debugAnnote = helpers.mkNullOrOption types.str ''
+      debug_annote = lib.nixvim.mkNullOrOption types.str ''
         Default annotation for debug items.
       '';
 
-      infoAnnote = helpers.mkNullOrOption types.str ''
+      info_annote = lib.nixvim.mkNullOrOption types.str ''
         Default annotation for info items.
       '';
 
-      warnAnnote = helpers.mkNullOrOption types.str ''
+      warn_annote = lib.nixvim.mkNullOrOption types.str ''
         Default annotation for warn items.
       '';
 
-      errorAnnote = helpers.mkNullOrOption types.str ''
+      error_annote = lib.nixvim.mkNullOrOption types.str ''
         Default annotation for error items.
       '';
 
-      priority = helpers.defaultNullOpts.mkUnsignedInt 50 ''
+      priority = defaultNullOpts.mkUnsignedInt 50 ''
         Order in which group should be displayed.
       '';
 
-      skipHistory = helpers.defaultNullOpts.mkBool true ''
+      skip_history = defaultNullOpts.mkBool true ''
         Whether progress notifications should be omitted from history.
       '';
+
+      update_hook = defaultNullOpts.mkNullable (with types; either rawLua (enum [ false ])) false ''
+        Called when an item is updated
+      '';
     };
   };
 in
-{
-  # TODO: deprecation warnings introduced 2023/12/22: remove in early February 2024
-  imports =
-    map
-      (
-        oldOption:
-        mkRemovedOptionModule
-          [
-            "plugins"
-            "fidget"
-            oldOption
-          ]
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "fidget";
+  packPathName = "fidget.nvim";
+  package = "fidget-nvim";
+
+  maintainers = [ lib.maintainers.HeitorAugustoLN ];
+
+  settingsOptions = {
+    progress = {
+      poll_rate =
+        defaultNullOpts.mkNullableWithRaw (with types; either ints.unsigned (enum [ false ])) 0
           ''
-            Nixvim: The fidget.nvim plugin has been completely rewritten. Hence, the options have changed.
-            Please, take a look at the updated documentation and adapt your configuration accordingly.
+            How and when to poll for progress messages.
+
+            Set to `false` to disable polling altogether. You can still manually pool
+            progress messages by calling |fidget.progress.poll|.
+          '';
+
+      suppress_on_insert = defaultNullOpts.mkBool false ''
+        Suppress new messages while in insert mode.
+      '';
 
-            > https://github.com/j-hui/fidget.nvim
+      ignore_done_already = defaultNullOpts.mkBool false ''
+        Ignore new tasks that don't contain a message.
```

Comment:
This doesn't look right. The old description and the name of the property don't seem to match up.

Reply:
Yeah, this was wrong indeed.